lib/diversity/component.rb

```diff
@@ -56,30 +56,26 @@ def initialize(spec, options)
       @options = DEFAULT_OPTIONS.keep_merge(options)
       @logger = @options[:logger]
 
-      schema = JsonSchemaCache[
-                 MASTER_COMPONENT_SCHEMA,
-                 { validate_spec: @options[:validate_spec] }
-               ]
-
       # Handle both Hash and String specs
       @raw = (spec.is_a?(String) ? parse_config(spec) : spec)
 
-      validation = schema.validate(@raw)
-      raise Diversity::Exception, "Configuration is not valid:\n" + validation.join("\n"), caller if
-        validation.length > 0
+      if (@options[:validate_spec])
+        schema = JsonSchemaCache[
+          MASTER_COMPONENT_SCHEMA,
+          { validate_spec: @options[:validate_spec] }
+        ]
+
+        validation = schema.validate(@raw)
+        raise Diversity::Exception,
+              "Configuration is not valid:\n" + validation.join("\n"), caller if
+          validation.length > 0
+      end
 
-      @checksum = Digest::SHA1.hexdigest(dump)
+      @checksum = "#{name}:#{version}"
       @assets = {}
       populate(@raw)
     end
 
-    # Returns a JSON dump of the component configuration
-    #
-    # @return [String]
-    def dump
```

Comment:
Fine. Feel free to remove everything we don't use at the moment. Then rerun the benchmarks and watch everything be just as slow. It's not unused methods that is slowing things down, it's the overall design that needs to overhauled to better match with our primary use case.

It's almost impossible to design a library correctly _before_ someone starts using it. One has to guess what *might* be needed, not just provide the things one needs for himself at the moment. If we intend only to serve our **own needs** with this library we shouldn't have a github project for it, we should have an internal project that only does exactly what *we* need. That *will* be fast - and useless for anyone else.

But like I said - feel free to merge this. Just don't ask me to readd the same stuff later.